bin/check-stale-results.rb

```diff
@@ -60,17 +61,26 @@ def humanize(secs)
     end.compact.reverse.join(' ')
   end
 
+  def get_uri(path)
+    protocol = (settings['api'].key?('protocol') ? settings['api']['protocol'] : 'http')
+    URI("#{protocol}://#{settings['api']['host']}:#{settings['api']['port']}#{path}")
+  end
+
   def api_request(method, path)
     unless settings.key?('api')
       raise 'api.json settings not found.'
     end
-    http = Net::HTTP.new(settings['api']['host'], settings['api']['port'])
-    req = net_http_req_class(method).new(path)
-    if settings['api']['user'] && settings['api']['password']
-      req.basic_auth(settings['api']['user'], settings['api']['password'])
+    uri = get_uri(path)
+    response = nil
```

Comment:
what does this serve? We assign it further down a little bit further to something else. In ruby you can assign vars on the fly no need to set them up beforehand.

Reply:
Ah, makes sense. Changed as per your suggestion.